src/test/app/AccountDelete_test.cpp

```diff
@@ -912,6 +912,373 @@ class AccountDelete_test : public beast::unit_test::suite
+            }
+
+            testcase("Expired credentials");
+            {
+                Account const john{"john"};
+
+                env.fund(XRP(10000), john);
+                env.close();
+
+                auto jv = credentials::create(john, carol, credType);
+                uint32_t const t = env.current()
+                                       ->info()
+                                       .parentCloseTime.time_since_epoch()
+                                       .count() +
+                    20;
+                jv[sfExpiration.jsonName] = t;
+                env(jv);
+                env.close();
+                env(credentials::accept(john, carol, credType));
+                env.close();
+                jv = credentials::ledgerEntry(env, john, carol, credType);
+                std::string const credIdx =
+                    jv[jss::result][jss::index].asString();
+
+                incLgrSeqForAccDel(env, john);
+
+                // credentials are expired
+                // john use credentials but can't delete account
+                env(acctdelete(john, alice),
+                    credentials::ids({credIdx}),
+                    fee(acctDelFee),
+                    ter(tecEXPIRED));
+                env.close();
+
+                {
+                    // check that expired credential object deleted
+                    auto jv =
+                        credentials::ledgerEntry(env, john, carol, credType);
+                    BEAST_EXPECT(
+                        jv.isObject() && jv.isMember(jss::result) &&
+                        jv[jss::result].isMember(jss::error) &&
+                        jv[jss::result][jss::error] == "entryNotFound");
+                }
+            }
+        }
+
+        {
+            testcase("Credentials feature disabled");
+            using namespace test::jtx;
+
+            Account const alice{"alice"};
+            Account const becky{"becky"};
+            Account const carol{"carol"};
+
+            Env env{*this, supported_amendments() - featureCredentials};
+            env.fund(XRP(100000), alice, becky, carol);
+            env.close();
+
+            // alice sets the lsfDepositAuth flag on her account.  This should
+            // prevent becky from deleting her account while using alice as the
+            // destination.
+            env(fset(alice, asfDepositAuth));
+            env.close();
+
+            // Close enough ledgers to be able to delete becky's account.
+            incLgrSeqForAccDel(env, becky);
+
+            auto const acctDelFee{drops(env.current()->fees().increment)};
+
+            // becky can't delete account as DepositPreauth object doesn't
+            // exists
+            env(acctdelete(becky, alice),
+                credentials::ids(
+                    {"098B7F1B146470A1C5084DC7832C04A72939E3EBC58E68AB8B579BA07"
+                     "2B0CECB"}),
+                fee(acctDelFee),
+                ter(tecNO_PERMISSION));
```

Comment:
Shouldn't this test result in `temDISABLED`, since the `CredentialIDs` field is provided but the `featureCredentials` amendment hasn't been enabled?

Reply:
fixed